opentelemetry-jaeger/src/exporter/agent.rs

```diff
@@ -40,12 +43,17 @@ pub(crate) struct AgentAsyncClientUDP {
     #[cfg(all(feature = "async-std", not(feature = "tokio")))]
     conn: async_std::net::UdpSocket,
     buffer_client: BufferClient,
+    max_packet_size: usize,
 }
 
 impl AgentAsyncClientUDP {
     /// Create a new UDP agent client
-    pub(crate) fn new<T: ToSocketAddrs>(host_port: T) -> thrift::Result<Self> {
-        let (buffer, write) = TBufferChannel::with_capacity(512).split()?;
+    pub(crate) fn new<T: ToSocketAddrs>(
+        host_port: T,
+        max_packet_size: Option<usize>,
+    ) -> thrift::Result<Self> {
+        let max_packet_size = max_packet_size.unwrap_or(UDP_PACKET_MAX_LENGTH);
```

Comment:
Sorry, but I still think this isn't quite right. Here, you default `max_packet_size` to `UDP_PACKET_MAX_LENGTH` which is still 65000, but the `UdpSocket::bind()` below binds on `0.0.0.0`, which is an IPv4 address. I think we can do better on this?

Reply:
What did you have in mind?

Reply:
Maybe try to connect to localhost via IPv6 and if that's doesn't work, then try to connect to localhost via IPv4? We could set the default max packet size based on the protocol

Reply:
Or just be conservative and restrict `max_packet_size` to 9216 by default since it can be configured anyway? (Maybe also rename `UDP_PACKET_MAX_LENGTH` to `DEFAULT_MAX_PACKET_SIZE`?)

Reply:
Hm looking around seems the other languages do 65,000 as well. e.g. [go](https://github.com/open-telemetry/opentelemetry-go/blob/9524ac095466565e01746600da3d83f27220a5da/exporters/trace/jaeger/agent.go#L30), [ruby](https://github.com/open-telemetry/opentelemetry-ruby/blob/dc109c7d851539864961a94d1510d67a10bbbac9/exporter/jaeger/README.md#how-can-i-configure-the-jaeger-exporter), [python](https://github.com/open-telemetry/opentelemetry-python/blob/18761d78e30a4293abd65a897129b57fbf0c0a15/exporter/opentelemetry-exporter-jaeger/src/opentelemetry/exporter/jaeger/send/thrift.py#L25), etc. Probably to be in sync with the [jaeger default](https://github.com/jaegertracing/jaeger/blob/5062366157fff65a052dc6ac6779e7621054248d/cmd/agent/app/servers/thriftudp/transport.go#L28-L31).

Reply:
To me that just looks like everybody is getting it wrong -- but I'm fine if you want to keep it to stay aligned.

Reply:
I'm ok with reducing if that is a common case, I know that macos will have the lower limit by default. Anyone have a good resource on different os defaults? If non-mac systems default to 65k we could just have a cfg on the const specifically for that.

Reply:
I think it will confuse ppl if the Rust implementation behaviors different from other implementations. So I incline to keep our limit here the same as others.

Reply:
Works for me if we want to merge this as is. Happy to update in a subsequent PR if we find a better heuristic.